src/main/java/com/org/checkr/service/CandidateService.java

```diff
@@ -0,0 +1,92 @@
+package com.org.checkr.service;
+
+import com.org.checkr.dto.CandidateDTO;
+import com.org.checkr.entity.Candidate;
+import com.org.checkr.repository.CandidateRepository;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CandidateService {
+    @PersistenceContext
+    private EntityManager entityManager;
+    private final CandidateRepository candidateRepository;
```

Comment:
here you are using entity manager and JPA repository both ,,use only once you can stick with JPA repo..Any speacfic reason to use entity manager?